source/ceylon/json/Visitor.ceylon

```diff
@@ -0,0 +1,84 @@
+"Callable interface used when traversing JSON data. 
+ 
+ It is the callers responsiblity to ensure the 
+ methods of this interface are called in a sequence 
+ that corresponds to well-formed JSON. 
+ For example, callers should never generate the calling
+ sequence `onKey()`, `onKey()`."
+by("Tom Bentley")
+shared interface Visitor {
```

Comment:
Well, initially we thought it would be silly to prefix every type in every module with the module's prefix, like XmlElement, JsonElement, HtmlElement, so we decided that in Ceylon one of the coding guideline is to remove modules name prefixes and resolve with import aliases. Perhaps that was a bad move, but it was a conscious one. We can't do it half, so either we conclude we should prefix types in every module, or we stick to the current plan. I'm open for discussing it, but I suppose in ceylon-spec.